wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -122,14 +128,24 @@ public WorkspaceImpl createWorkspace(
 
   @Traced
   public WorkspaceImpl createWorkspace(
-      Devfile devfile, String namespace, Map<String, String> attributes)
+      Devfile devfile,
+      String namespace,
+      Map<String, String> attributes,
+      FileContentProvider contentProvider)
       throws ServerException, NotFoundException, ConflictException, ValidationException {
     TracingTags.STACK_ID.set(() -> attributes.getOrDefault("stackId", "no stack"));
 
     requireNonNull(devfile, "Required non-null devfile");
     requireNonNull(namespace, "Required non-null namespace");
     validator.validateAttributes(attributes);
 
+    try {
+      devfileIntegrityValidator.validateDevfile(devfile);
```

Comment:
Seems we're doing it twice: https://github.com/eclipse/che/pull/13472/files#diff-7d12839ec56888bc72a7c94a083977c0R121

Reply:
Good catch!